src/main/scala/beam/router/BeamSkimmer.scala

```diff
@@ -372,9 +394,147 @@ class BeamSkimmer @Inject()(val beamConfig: BeamConfig) extends IterationEndsLis
           }
         }
       }
+
     writer.close()
   }
 
+  def writeTravelTimeObservedVsSimulated(event: IterationEndsEvent): Unit = {
+    val uniqueModes = List(CAR)
+    val uniqueTimeBins = (0 to 23)
+
+    val dummyId = Id.create("NA", classOf[BeamVehicleType])
+
+    val filePathObservedVsSimulated = event.getServices.getControlerIO.getIterationFilename(
+      event.getServices.getIterationNumber,
+      "tazODTravelTimeObservedVsSimulated.csv.gz"
+    )
+    val writerObservedVsSimulated = IOUtils.getBufferedWriter(filePathObservedVsSimulated)
+    writerObservedVsSimulated.write("fromTAZId,toTAZId,hour,timeSimulated,timeObserved")
+    writerObservedVsSimulated.write("\n")
+
+    val fjson = new FeatureJSON
```

Comment:
It still looks pretty big method. I'd suggest you to split it to two extra methods. First one would read and return `CensusTractHash`, the second one would use `CensusTractHash` and file path to CSV and return `observedTravelTimes`. My concern here is that method `writeTravelTimeObservedVsSimulated` does a lot of things. It has a name `write`, but it reads and prepare the data. Thanks.

Reply:
@REASY totally agree. Fixing. Thank you.